src/ComboBox/index.jsx

```diff
@@ -418,11 +418,11 @@ export default class ComboBox extends Component
             id,
             isOpen,
             searchValue,
-            selectedOption,
+            value,
         } = this.state;
 
-        const optionVal = getOption( selectedOption, flatOptions ) ?
-            getOption( selectedOption, flatOptions ).text : undefined;
+        const optionVal = getOption( value, flatOptions ) ?
+            getOption( value, flatOptions ).text : undefined;
```

Comment:
Not strictly related to this issue but... while we’re here... 😄 

Can we change this:
```
const optionVal = getOption( value, flatOptions ) ?
            getOption( selectedOption, flatOptions ).text : undefined;
```
to something like this:
```
const option = getOption( value, flatOptions );
const optionVal = option ? option.text : undefined;
```

(Seems pretty wasteful to call `getOption` twice)